inst/chunks/fun_scale_time_err_t.stan

```diff
@@ -0,0 +1,31 @@
+  /* scale and correlate time-series residuals
+   * aware of time variable (for skipped time points)
+   * Args:
+   *   zerr: standardized and independent residuals
+   *   sderr: standard deviation of the residuals
+   *   chol_cor: cholesky factor of the correlation matrix
+   *   n_times: number of time points in each group
+   *   begin: the first observation in each group
+   *   end: the last observation in each group
+   *   times: vector of time points
+   * Returns:
+   *   vector of scaled and correlated residuals
+   */
+   vector scale_time_err_t(vector zerr, real sderr, matrix chol_cor,
+                           int[] n_times, int[] begin, int[] end, int[] times) {
+     vector[rows(zerr)] err;
+     int temp_times[max(n_times)];
+     for (i in 1:size(n_times)) {
+       // Set up the indexing vector
+       temp_times[1] = 1;
+       for (j in 1:(n_times[i]-1)) {
+         temp_times[j+1] = temp_times[j] +
```

Comment:
can you explain to me what the temp_times are doing to make sure I understand correctly?

Reply:
I see. I think Stan will not allow you to edit input variables if I remember correctly, so using something like temp_times (or a shorter name to make lines not too long) is fine. Just wanted to understand what the purpose of this was.

Reply:
Just a side note, that I will eventually want to allow missing time points in all of the correlation structures, so I will likely build onto your implementation in the future and role it out to all the correlation structures.